src/sniffer/pkg/sniffer/sniffer.go

```diff
@@ -2,145 +2,112 @@ package sniffer
-type capturesMap map[string]map[string]time.Time
-
 type Sniffer struct {
-	capturedRequests capturesMap
-	socketScanner    *socketscanner.SocketScanner
-	lastReportTime   time.Time
-	mapperClient     mapperclient.MapperClient
+	dnsSniffer           *collectors.DNSSniffer
+	socketScanner        *collectors.SocketScanner
+	lastReportTime       time.Time
+	lastHostsRefreshTime time.Time
+	mapperClient         mapperclient.MapperClient
 }
 
 func NewSniffer(mapperClient mapperclient.MapperClient) *Sniffer {
 	return &Sniffer{
-		capturedRequests: make(capturesMap),
-		socketScanner:    socketscanner.NewSocketScanner(mapperClient),
-		lastReportTime:   time.Now(),
-		mapperClient:     mapperClient,
+		dnsSniffer:    collectors.NewDNSSniffer(ipresolver.NewProcFSIPResolver()),
+		socketScanner: collectors.NewSocketScanner(),
+		mapperClient:  mapperClient,
 	}
 }
 
-func (s *Sniffer) HandlePacket(packet gopacket.Packet) {
-	captureTime := detectCaptureTime(packet)
-	ipLayer := packet.Layer(layers.LayerTypeIPv4)
-	dnsLayer := packet.Layer(layers.LayerTypeDNS)
-	if dnsLayer != nil && ipLayer != nil {
-		ip, _ := ipLayer.(*layers.IPv4)
-		dns, _ := dnsLayer.(*layers.DNS)
-		if dns.OpCode == layers.DNSOpCodeQuery && dns.ResponseCode == layers.DNSResponseCodeNoErr {
-			for _, answer := range dns.Answers {
-				// This is the DNS Answer, so the Dst IP is the pod IP
-				if answer.Type != layers.DNSTypeA && answer.Type != layers.DNSTypeAAAA {
-					continue
-				}
-				s.addCapturedRequest(ip.DstIP.String(), string(answer.Name), captureTime)
-			}
-		}
+func (s *Sniffer) reportCaptureResults(ctx context.Context) error {
+	results := s.dnsSniffer.CollectResults()
+	if len(results) == 0 {
+		logrus.Debugf("No captured sniffed requests to report")
+		return nil
 	}
-}
 
-func detectCaptureTime(packet gopacket.Packet) time.Time {
-	captureTime := packet.Metadata().CaptureInfo.Timestamp
-	if captureTime.IsZero() {
-		return time.Now()
-	}
-	return captureTime
-}
+	timeoutCtx, cancelFunc := context.WithTimeout(ctx, viper.GetDuration(config.CallsTimeoutKey))
+	defer cancelFunc()
 
-func (s *Sniffer) addCapturedRequest(srcIp string, destDns string, seenAt time.Time) {
-	if _, ok := s.capturedRequests[srcIp]; !ok {
-		s.capturedRequests[srcIp] = make(map[string]time.Time)
+	logrus.Infof("Reporting captured requests of %d clients to Mapper", len(results))
+	err := s.mapperClient.ReportCaptureResults(timeoutCtx, mapperclient.CaptureResults{Results: results})
+	if err != nil {
+		return err
 	}
-	s.capturedRequests[srcIp][destDns] = seenAt
+	return nil
 }
 
-func (s *Sniffer) ReportCaptureResults(ctx context.Context) error {
-	s.lastReportTime = time.Now()
-	if len(s.capturedRequests) == 0 {
-		logrus.Debugf("No captured requests to report")
+func (s *Sniffer) reportSocketScanResults(ctx context.Context) error {
+	results := s.socketScanner.CollectResults()
+	if len(results) == 0 {
+		logrus.Debugf("No scanned tcp connections to report")
```

Comment:
Let's take this out side the report function and put it just below `refreshHostMapping` and before the report method. Would be more clear to the user where we handle both reading from DNS and Sockets info

Reply:
Reading sockets info does happen below the RefreshHostMapping. This function just converts the internal structures to the API's schema

Reply:
I see, cool